src/Apps/Order/Routes/Shipping2/Utils/shippingUtils.ts

```diff
@@ -135,11 +135,61 @@ export const getDefaultUserAddress = (
     if (!filterCountries) return true
     return filterCountries.includes(node.country)
   })
+
+  if (shippableAddresses.length === 0) {
+    return null
+  }
+
   return (
     shippableAddresses.find(node => node.isDefault) || shippableAddresses[0]
   )
 }
 
+export const getInitialShippingValues = (
+  savedAddresses: SavedAddressType[],
+  defaultCountry,
+  filterCountries?: string[]
+): ShipValues => {
+  const defaultUserAddress = getDefaultUserAddress(
+    savedAddresses,
+    filterCountries
+  )
+
+  // The default ship-to address should be the first one that
+  // can be shipped-to, preferring the default address if it exists.
+  const attributesFromDefaultAddress: ShipValues["attributes"] = addressWithFallbackValues(
+    defaultUserAddress
+  )
+  if (defaultUserAddress) {
+    return {
+      fulfillmentType: FulfillmentType.SHIP,
+      attributes: attributesFromDefaultAddress,
+      meta: {
+        saveAddress: false,
+        addressVerifiedBy: null,
+      },
+    }
+  }
+
+  // The user doesn't have a valid ship-to address, so we'll return empty values.
+  // TODO: This doesn't account for matching the saved address id
```

Comment:
This TODO is confusing to me. But as long as you know what to do here? Maybe if it's important we could create an issue to describe what needed to be done here and link it here if this refactor is important.

Does this say that if we reload the page we are not setting it to the address that was previously saved on the server and instead to the first one that fits?

Reply:
🔪